lib/plugins/aws/package/compile/events/msk/index.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+
+const getMskClusterNameToken = require('./getMskClusterNameToken');
+
+class AwsCompileMSKEvents {
+  constructor(serverless) {
+    this.serverless = serverless;
+    this.provider = this.serverless.getProvider('aws');
+
+    this.hooks = {
+      'package:compileEvents': this.compileMSKEvents.bind(this),
+    };
+
+    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'msk', {
+      type: 'object',
+      properties: {
+        arn: {
+          oneOf: [
+            { $ref: '#/definitions/awsArnString' },
+            { $ref: '#/definitions/awsCfImport' },
+            { $ref: '#/definitions/awsCfRef' },
+          ],
+        },
+        batchSize: {
+          type: 'number',
+          minimum: 1,
+          maximum: 10000,
+        },
+        enabled: {
+          type: 'boolean',
+        },
+        startingPosition: {
+          type: 'string',
+          enum: ['LATEST', 'TRIM_HORIZON'],
+        },
+        topic: {
+          type: 'string',
+        },
+      },
+      additionalProperties: false,
+      required: ['arn', 'topic'],
+    });
+  }
+
+  compileMSKEvents() {
+    this.serverless.service.getAllFunctions().forEach(functionName => {
+      const functionObj = this.serverless.service.getFunction(functionName);
+      const cfTemplate = this.serverless.service.provider.compiledCloudFormationTemplate;
+
+      if (functionObj.events) {
```

Comment:
`functionObj.events` is guaranteed by normalization at: https://github.com/serverless/serverless/blob/feece9a2ec5be0f49af7147b84bed76e9ba50155/lib/classes/Service.js#L182-L183

So it's good to assume it's always there (in many other places we rely on that assumption)